File: 3-longest-substring-without-repeating-characters/3-longest-substring-without-repeating-characters.py
def contain_duplicate(s):
    hashmap = {}
    for char in s:
        if char not in hashmap:
            hashmap[char] = 1
        else:
            return True
    return False

class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        if s == "":
            return 0
        
        ret = 1
        
        for window_size in range(2, len(s) + 1):
            stop = True
            
            for i in range(len(s) - window_size + 1):
                window = s[i:i + window_size]
                if not contain_duplicate(window):
                    stop = False
                    break
            
            if stop:
                break
            else:
                ret = window_size
            
        return ret

File: 3-longest-substring-without-repeating-characters/3-longest-substring-without-repeating-characters.java
import java.util.HashMap;

class Solution {
    public boolean containDuplicate(String s) {
        HashMap<Character, Integer> map = new HashMap<>();
        for (int i = 0; i < s.length(); i++) {
            if (!map.containsKey(s.charAt(i))){
                map.put(s.charAt(i), 1);
            } else {
                return true;
            }
        }
        return false;
    }
    
    public int lengthOfLongestSubstring(String s) {
        if (s.equals("")) return 0;
        
        int ret = 1;
        for (int windowSize = 2; windowSize <= s.length(); windowSize++) {
            boolean stop = true;
            for (int i = 0; i <= s.length() - windowSize; i++) {
                String window = s.substring(i, i + windowSize);
                if (!containDuplicate(window)) {
                    stop = false;
                    break;
                }
            }
            
            if (stop) {
                break;
            } else {
                ret = windowSize;
            }
        }
        return ret;
    }
}